tests/Bridge/Doctrine/Orm/Extension/FilterEagerLoadingExtensionTest.php

```diff
@@ -123,10 +126,150 @@ public function testApplyCollection()
+        $filterEagerLoadingExtension->applyToCollection($qb, $queryNameGenerator->reveal(), DummyCar::class, 'get');
+
+        $expected = <<<SQL
+SELECT o, count(o.id) as counter 
+FROM ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\DummyCar o 
+LEFT JOIN o.colors colors WHERE o 
+IN(
+  SELECT o_2 FROM ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\DummyCar o_2 
+  LEFT JOIN o_2.colors colors_2 
+  WHERE o_2.colors = :foo
+) 
+GROUP BY o.colors HAVING counter > 3 
+ORDER BY o.colors ASC
+SQL;
+
+        $this->assertEquals($this->toDQLString($expected), $qb->getDQL());
+    }
+
+    public function testCompositeIdentifiers()
+    {
+        $resourceMetadataFactoryProphecy = $this->prophesize(ResourceMetadataFactoryInterface::class);
+        $resourceMetadataFactoryProphecy->create(CompositeRelation::class)->willReturn(new ResourceMetadata(CompositeRelation::class));
+
+        $classMetadata = new ClassMetadataInfo(CompositeRelation::class);
+        $classMetadata->isIdentifierComposite = true;
+        $classMetadata->identifier = ['item', 'label'];
+
+        $em = $this->prophesize(EntityManager::class);
+        $em->getExpressionBuilder()->shouldBeCalled()->willReturn(new Expr());
+        $em->getClassMetadata(CompositeRelation::class)->shouldBeCalled()->willReturn($classMetadata);
+
+        $qb = new QueryBuilder($em->reveal());
+
+        $qb->select('o')
+            ->from(CompositeRelation::class, 'o')
+            ->innerJoin('o.compositeItem', 'item')
+            ->innerJoin('o.compositeLabel', 'label')
+            ->where('item.field1 = :foo')
+            ->setParameter('foo', 1);
+
+        $queryNameGenerator = $this->prophesize(QueryNameGeneratorInterface::class);
+        $queryNameGenerator->generateJoinAlias('item')->shouldBeCalled()->willReturn('item_2');
+        $queryNameGenerator->generateJoinAlias('label')->shouldBeCalled()->willReturn('label_2');
+        $queryNameGenerator->generateJoinAlias('o')->shouldBeCalled()->willReturn('o_2');
+
+        $filterEagerLoadingExtension = new FilterEagerLoadingExtension($resourceMetadataFactoryProphecy->reveal(), true);
+        $filterEagerLoadingExtension->applyToCollection($qb, $queryNameGenerator->reveal(), CompositeRelation::class, 'get');
+
+        $expected = <<<SQL
+SELECT o 
+FROM ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\CompositeRelation o 
+INNER JOIN o.compositeItem item 
+INNER JOIN o.compositeLabel label 
+WHERE o.item IN(
+    SELECT IDENTITY(o_2.item) FROM ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\CompositeRelation o_2 
+    INNER JOIN o_2.compositeItem item_2 
+    INNER JOIN o_2.compositeLabel label_2 
+    WHERE item_2.field1 = :foo
+) AND o.label IN(
+    SELECT IDENTITY(o_2.label) FROM ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\CompositeRelation o_2 
+    INNER JOIN o_2.compositeItem item_2 
+    INNER JOIN o_2.compositeLabel label_2 
+    WHERE item_2.field1 = :foo
+)
+SQL;
```

Comment:
Oh, but I think the tests are confusing, because there's no eager loading going on (how do we tell?). If so, we can skip query rewriting in such cases, right?